src/test/java/use_case/add_thread/AddThreadInteractorTest.java

```diff
@@ -0,0 +1,369 @@
+package use_case.add_thread;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import entity.Message;
+import entity.Thread;
+import entity.ThreadFactory;
+
+/**
+ * Class for testing the Add Thread Interactor.
+ */
+class AddThreadInteractorTest
+{
+    private ThreadFactory threadFactory;
+
+    private AddThreadInputData validInputData;
+    private AddThreadInputData invalidInputDataEmptyName;
+    private AddThreadInputData invalidInputDataNameTooLong;
+    private AddThreadInputData invalidInputDataNameTooShort;
+    private AddThreadInputData invalidInputDataEmptyUsers;
+    private AddThreadInputData invalidInputDataPoorlyFormattedUsers;
+
+    @BeforeEach
+    void setUp()
+    {
+        // Setup mock thread factory
+        threadFactory = new ThreadFactory();
+
+        // Setup valid input data
+        validInputData = new AddThreadInputData(
+                "Valid Thread", "user1,user2,user3", "TestJoe");
+
+        // Setup invalid input data
+        invalidInputDataEmptyName = new AddThreadInputData(
+                "", "user1,user2", "TestJoe");
+        invalidInputDataNameTooLong = new AddThreadInputData(
+                "A".repeat(Thread.THREAD_NAME_MAX_LENGTH + 1),
+                "user1,user2", "TestJoe");
+        invalidInputDataNameTooShort = new AddThreadInputData(
+                "A", "user1,user2", "TestJoe");
+        invalidInputDataEmptyUsers = new AddThreadInputData(
+                "Valid Thread", "", "TestJoe");
+        invalidInputDataPoorlyFormattedUsers = new AddThreadInputData(
+                "Valid Thread", "user1,,user2", "TestJoe");
+    }
+
+    @Test
+    void testExecuteValidInputSuccess()
+    {
+        // create mock DAO for this specific test
+        AddThreadThreadDataAccessInterface mockServer = new InMemoryThreadDataAccess();
+
+        // create mock presenter for this specific test
+        AddThreadOutputBoundary addThreadPresenter = new AddThreadOutputBoundary()
+        {
+            @Override
+            public void prepareSuccessView(AddThreadOutputData outputData, String successMessage)
+            {
+                // TODO: assert the correct successes here
+                fail("Success is unexpected here");
+            }
+
+            @Override
+            public void prepareFailView(String errorMessage)
+            {
+                fail("Failure is unexpected here");
+            }
+
+            @Override
+            public void switchToThreadsView()
```

Comment:
-  Not certain if you need to test the switch to threads view method - maybe we can leave it empty if you still have to define the method here.
-